airflow-core/src/airflow/ui/src/components/PoolBar.tsx

```diff
@@ -42,13 +42,28 @@ export const PoolBar = ({
         const slotValue = pool[key];
         const flexValue = slotValue / totalSlots || 0;
 
+        // Hide empty segments
         if (flexValue === 0) {
           return undefined;
         }
 
+        // NEW: If this segment represents "deferred_slots" and the current pool
+        // is configured NOT to include deferred in occupied slots, hide this chip.
+        // (We only have the flag on real pools; aggregated "Slots" objects don't carry it)
+        const isDeferredKey = key === "deferred_slots";
+
+        if (
+          isDeferredKey &&
+          "include_deferred" in pool && // type guard: only PoolResponse has this
```

Comment:
We are in a `PoolBar` component -- what else could we be called with if not PoolResponse?